service/api_logic/user_logic.py

```diff
@@ -334,18 +281,14 @@ async def refresh_tokens(self):
         if not self._refresh_dal.verify_nonce(identity, token_nonce):
             raise InvalidRefreshTokenError()
 
-        new_access_token, new_refresh_token = await self.create_new_access_and_refresh_tokens(identity, refresh=True)
+        new_access_token, new_refresh_token = await self.create_tokens(identity, refresh=True)
 
         new_nonce = self.generate_nonce()
         self._refresh_dal.update_refresh_token(identity, new_refresh_token, new_nonce)
 
-        response = jsonify({
-            "access_token": new_access_token,
-            "refresh_token": new_refresh_token
-        })
-    
-        set_access_cookies(response, new_access_token)
-        set_refresh_cookies(response, new_refresh_token)
+        user = await self._user_dal.get_user_by_email(identity)
+        
+        return new_access_token, new_refresh_token, user
```

Comment:
Why do you keep returning user everywhere like this?

Even if we truly need it (even though no changes to user were made, so we actually shouldn't), we can make it in a somewhat better way. Call a method in one place and not in 3 or 4

That violates incapsulation